RemoteSigner/Function.cs

```diff
@@ -62,20 +62,14 @@ public class Function
             var kmsClient = new AmazonKeyManagementServiceClient();
 #endif
 
+            var network = ParseNetwork(requestBody.Network);
 
-            var network = requestBody.Network.ToUpper() switch
-            {
-                "REGTEST" => Network.RegTest,
-                "MAINNET" => Network.Main,
-                "MAIN" => Network.Main, //NBitcoin uses "Main" as its network name for mainnet
-                "TESTNET" => Network.TestNet,
-                _ => throw new ArgumentException("Network not recognized")
-            };
-            
             Console.WriteLine($"Network: {network}");
 
             if (PSBT.TryParse(requestBody.Psbt, network, out var parsedPSBT))
             {
+                parsedPSBT.AssertSanity();
+                
                 foreach (var psbtInput in parsedPSBT.Inputs)
```

Comment:
I see the for loop here, why wasn't signing all inputs?

Reply:
it was wrong to use  `parsedPSBT.SignWithKeys(key);` because that already has a loop, and also there was a condition where it was signing all the inputs or doing loops in loops, I did TDD and this is how it worked.